rapid7/metasploit-framework
# Windows local TokenMagic exploit module

Exploit module to leverage [TokenMagic](https://github.com/FuzzySecurity/PowerShell-Suite/blob/master/UAC-TokenMagic.ps1), which duplicates the token of elevated process and does some magic in order to spawn a new process with a high integrity level. Successful exploitation results in code execution as SYSTEM. This module can leverage tokenmagic in two ways; it can use the high IL to create and spawn malicious service or to write a dll to system32 that is known to be vulnerable to hijacking. 

The module has been tested against the following targets:
Service method: Windows 7, 8.1, Windows 10 (1511, 1803)
DLL method: Windows 10 (1703, 1803)

## Verification Steps
 1. Start msfconsole
 2. Get at least a meterpreter session on a Windows target
 3. Do: `use exploit/windows/local/tokenmagic`
 4. Set the `METHOD` of exploitation, either `DLL` or `SERVICE`
 5. Set the `LHOST`, `SESSION` and `PAYLOAD` options
 6. Do: `run`
 7. You should get a shell, the exploitation process should be fairly instantaneous when using the service method, the DLL method takes some time to trigger and receive a shell, up to 15 minutes.
